### PR TITLE
feat(provider/kubernetes): support for kubectl server-side-apply stra…

### PR DESCRIPTION
**Pull request by aman-agrawal**
**Jira** : https://devopsmx.atlassian.net/browse/OP-21866
**Testing** : compile succesful with java11, no UT failing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,4 +34,7 @@ e458cd21087f54b0a9ece3dbe967edfafe98d001 | Yugandharkumar | Create commits-prese
 986aceb68240b2fb93bd923d78235133a619e1fb | Yugandharkumar | adding tzdata-java | 2023-08-25 
 
 
-a81f01042740e1d03b403c8f66b274fcd17370df | Yugandharkumar | Rename Dockerfile-sleep to Dockerfile | 2024-05-20 
\ No newline at end of file
+a81f01042740e1d03b403c8f66b274fcd17370df | Yugandharkumar | Rename Dockerfile-sleep to Dockerfile | 2024-05-20
```
Comment (j-sandy) [inline README.md:37]: Is this change intentional ?

Comment (j-sandy) [reply README.md:37]: First of all I feel the workflow is misbehaving, and putting the commits in ReadMe file. This has to be checked by @yugaa22 , but since this is a one off case with specific repo, which is not going to be used as a regular repo, we can ignore to rectify the workflow.

Comment (aman-agrawal) [reply README.md]: README changes discarded.

**Review by j-sandy — APPROVED** [review]
LGTM !
